app/models/token.rb

```diff
@@ -0,0 +1,26 @@
+class Token < ActiveRecord::Base
+  belongs_to :user
+  has_and_belongs_to_many :scopes
+
+  before_validation :setup, on: :create
+
+  validates :token, presence: true, uniqueness: true
+
+  scope :valid, ->(time) { where("expires_at >= ?", time) }
```

Comment:
I still haven't figured out how we are going to get rid of expired tokens. I'm thinking another background task like https://github.com/diaspora/diaspora/pull/6147. If it still makes sense after that, I'll refactor this as so.